aio_pika/robust_connection.py

```diff
@@ -134,9 +134,18 @@ def channel(self, channel_number: int = None,
         for number, channel in self._channels.items():
             try:
                 await channel.on_reconnect(self, number)
-            except CONNECTION_EXCEPTIONS:
+            except CONNECTION_EXCEPTIONS as e:
                 log.exception('Open channel failure')
-                await self.close()
+
+                if self.connection is not None:
+                    await asyncio.gather(
+                        self.connection.close(e), return_exceptions=True
+                    )
+                    self.connection = None
+
+                # The channel has not been opened successfully.
+                # Staying positive and just retry later.
+                await asyncio.sleep(self.reconnect_interval)
```

Comment:
I guess this has no effect.. Maybe we should get rid of `_on_reconnect` method and rewrite `reconnect` method as follows:
```python
    async def reconnect(self):
        if self.is_closed:
            return

        while True:
            try:
                await super().connect()
            except CONNECTION_EXCEPTIONS:
                log.exception('Connection attempt error')
                await asyncio.sleep(self.reconnect_interval)
            else:
                try:
                    for number, channel in self._channels.items():
                        await channel.on_reconnect(self, number)
                except CONNECTION_EXCEPTIONS as e:
                    log.exception('Open channel failure')
                    if self.connection is not None:
                        await self.close(e)
                        self.connection = None
                else:
                    break
        self._reconnect_callbacks(self)
```
What do you think?

Reply:
I try this, it's broken. I don't want to refactor this part now, just fix production issue.

Reply:
OK, but what this sleep gives here? I think the bug is `await self.close()` which prevents `reconnect` method to connect again (because of `self._closed` flag which you mentioned earlier).

Reply:
If you meant waiting before reconnect attempt then there is already sleep at `_on_connection_close` callback..

Reply:
Yeah. You are right. After rewrite looks better.